Add page on Brick/REC modeling with 223 (#30)
open223/docs.open223.info: guides/brick-rec.md

# Using 223 with Brick and RealEstateCore

The [Brick](https://brickschema.org) and [RealEstateCore](https://realestatecore.io) ontologies can be used to augment 223 models with a richer vocabulary of concepts.

> There are a few reasons to consider doing this:

Reviewer: ```suggestion
The [Brick](https://brickschema.org) and [RealEstateCore](https://realestatecore.io) ontologies can be used to augment 223 models with a richer vocabulary of concepts, which offer the following benefits.

@prefix unit: <http://qudt.org/vocab/unit/> .
@prefix brick: <https://brickschema.org/schema/Brick#> .

bldg:damper a s223:Damper ;
    s223:cnx bldg:damper-out .

bldg:sensor a s223:Sensor ;
    s223:hasObservationLocation bldg:damper-out ;
    s223:hasPhysicalLocation bldg:damper ;
    s223:observes bldg:air-temp .

bldg:damper-out a s223:OutletConnectionPoint ;
    s223:cnx bldg:out-connection ;
    s223:hasMedium s223:Medium-Air ;
    s223:hasProperty bldg:air-temp .

bldg:air-temp a s223:QuantifiableObservableProperty,
                brick:Supply_Air_Temperature_Sensor ;
    qudt:hasQuantityKind qudtqk:Temperature ;
    s223:hasAspect s223:Role-Supply ;
    qudt:hasUnit unit:DEG_C .
```

By stating that the `bldg:air-temp` entity is also a `brick:Supply_Air_Temperature_Sensor`, Brick-based applications can more easily find the data they are looking for.
223 requires multiple annotations be combined in order to model this concept.

The Brick class also clearly states the role of the entity.
The use of the QUDT annotations and Brick types means that this 223 model is also a valid Brick model!

Using Brick can simplify some queries against the model.

> Without Brick, finding the `bldg:air-temp` entity would require this query:

Reviewer: ```suggestion
For example, finding the `bldg:air-temp` entity without Brick would require this query:
```

```sparql
PREFIX bldg: <urn:ex/>
PREFIX s223: <http://data.ashrae.org/standard223#>
PREFIX qudt: <http://qudt.org/schema/qudt/>
PREFIX qudtqk: <http://qudt.org/vocab/quantitykind/>
SELECT ?damper ?temp WHERE {
    ?damper a s223:Damper ;
        s223:hasProperty ?temp.
    ?temp a s223:QuantifiableObservableProperty ;
        qudt:hasQuantityKind qudtqk:Temperature ;
        s223:hasAspect s223:Role-Supply .
}
```

> whereas with Brick, the query can be simplified to

Reviewer: ```suggestion
With Brick, the query above can be simplified to the following where the `brick:hasPoint` relationship is inferred automatically.
```

```sparql
PREFIX bldg: <urn:ex/>
PREFIX s223: <http://data.ashrae.org/standard223#>
PREFIX brick: <https://brickschema.org/schema/Brick#>

SELECT ?damper ?temp WHERE {
    ?damper a s223:Damper ;
        brick:hasPoint ?temp .
    ?temp a brick:Supply_Air_Temperature_Sensor .
}
```

> The `brick:hasPoint` relationship is inferred automatically.

Reviewer: ```suggestion
```

## Using Brick `Equipment`s with 223

Brick's extensive `Equipment` classes can also be used with 223.

> Any 223 `Equipment` can also be annotated with a Brick `Equipment` class, for example:

Reviewer: ```suggestion
Any 223 `Equipment` can also be annotated with a Brick `Equipment` class as shown in the example below. 
```
